src/components/header/excel_popup/ExcelPopup.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import "./excelPopup.scss";
+import PropTypes from 'prop-types';
+import { withRouter } from "react-router-dom";
+import { RouterEnum } from '../../../enums/router-enum';
+import GeneratorAndDownloader from '../../Excelerator/generator_and_downloader/GeneratorAndDownloader';
+import { GET_ALL_USERS_FOR_EXCEL_QUERY, GET_All_USERS, SYNC_USERS } from '../../../graphql/UserGql';
+import Uploader from '../../Excelerator/Uploader/Uploader';
+import { EntitySchemaForExcel } from '../../Excelerator/Excelerator';
+import { useLazyQuery } from '@apollo/react-hooks';
+import { GET_ENTITY_SCHEMA_FOR_EXCEL } from '../../../graphql/SchemaGql';
+import Saver from '../../Excelerator/Saver/Saver';
+
+
+ExcelPopup.propTypes = {
+    closePopup: PropTypes.func.isRequired,
+    history: PropTypes.shape({
+        listen: PropTypes.func.isRequired,
+        location: PropTypes.shape({
+            pathname: PropTypes.string.isRequired
+        }).isRequired
+    }).isRequired
+}
+
+function ExcelPopup({ closePopup, history }) {
+    const [getEntitySchemaFromGQL, { error: getTypeError, loading: getTypeLoading, data: entitySchemaFromGQL }] = useLazyQuery(GET_ENTITY_SCHEMA_FOR_EXCEL);
+
+    const [isSaveDisabled, setIsSaveDisabled] = useState(true);
+
+    const [entityName, setEntityName] = useState("");
+    const [getAllDataQueryForBuildingExcel, setGetAllDataQueryForBuildingExcel] = useState();
+    const [allDataQueryName, setallDataQueryName] = useState("");
+    const [getAllDataQueryForUpdatingState, setGetAllDataQueryForUpdatingState] = useState();
+    const [syncMutation, setSyncMutation] = useState();
+
+    useEffect(() => {
+        let entityName;
+
+        switch (history.location.pathname) {
+            case RouterEnum.USERS: {
+                entityName="User";
+                setEntityName(entityName);
+
+                //for gernerator-and-downloader comp
+                setGetAllDataQueryForBuildingExcel(GET_ALL_USERS_FOR_EXCEL_QUERY);
+                setallDataQueryName("allUsers");
+
+                // for saver comp
+                setGetAllDataQueryForUpdatingState(GET_All_USERS);
```

Comment:
This is the query called to update the case in the saver component? if it is maybe we can call it updateCacheQuery